api/filter_dimensions.go

```diff
@@ -87,12 +83,12 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 		)
 		return
 	}
-
-	if err := api.isValidDatasetDimensions(ctx, v, []model.Dimension{req.Dimension}, filter.PopulationType); err != nil {
+	dimensions, _, err := api.ValidateCantabularFilter(v, []model.Dimension{req.Dimension}, filter.PopulationType)
+	if err != nil {
 		api.respond.Error(ctx, w, statusCode(err), err)
 		return
 	}
-
+	finalDim := dimensions[0]
```

Comment:
can we please add some comment to help understand 'final dimension'

Reply:
I just didn't want to reference dimensions[0] all the time, not sure if it is worth a comment